### PR TITLE
feat: Add basic changelog generator tool

### PR DESCRIPTION
**Pull request by tylerbutler**
This tool is used to transform changesets into CHANGELOG.md entries. The new project is private and in the @fluid-internal scope. It's used as a `file:` dev dependency in the repo root.

It uses the extensibility that the default changesets tools provide, which is documented here: <https://github.com/changesets/changesets/blob/main/docs/modifying-changelog-format.md>

In essence, it takes all the individual changeset files and updates the CHANGELOG.md file for each package with the changesets that apply to it. Release engineers will use this when doing minor and major releases.

Unfortunately the changelog APIs are not well documented, so this tool builds on top of another formatter, [changesets-format-with-issue-links](https://github.com/spautz/changesets-changelog-format). The only changes we've made to that formatter is to ignore changelog entries that are only due to dependency updates. The changelog files will then be fixed up using the command in PR #15949.

Related to [AB#3975](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/3975)

### CODE REVIEW

File: .github/workflows/data/patch-changelog.hbs
Comment (tylerbutler) [inline .github/workflows/data/patch-changelog.hbs:1]: The changeset tools can't handle random files inside the .changeset folder.

File: build-tools/packages/changelog-generator/.eslintrc.cjs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+		// This package uses interfaces and types that are not exposed directly by npm-check-updates.
+		// We also call commands' run method directly in some cases, so these are all excluded.
+		"import/no-internal-modules": [
+			"error",
+			{
+				allow: ["npm-check-updates/build/src/types/**", "**/commands/**"],
+			},
+		],
+	},
```
Comment (tylerbutler) [inline build-tools/packages/changelog-generator/.eslintrc.cjs:25]:
```suggestion
```

File: build-tools/packages/changelog-generator/package.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	"author": "Microsoft and contributors",
+	"main": "./dist/index.js",
+	"files": [
+		"dist/",
+		"src/",
+		"LICENSE",
+		"*.md",
+		"*.cjs",
+		"*.json"
+	],
```
Comment (tylerbutler) [inline build-tools/packages/changelog-generator/package.json:21]:
```suggestion
	"files": [
		"dist/",
		"LICENSE",
		"*.md",
		"*.cjs",
		"*.json"
	],
```

File: build-tools/packages/changelog-generator/src/test/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+	"extends": "@fluidframework/build-common/ts-common-config.json",
+	"compilerOptions": {
+		"declaration": true,
+		"importHelpers": true,
+		"declarationMap": false,
+		"rootDir": "./",
+		"outDir": "../../lib/test",
```
Comment (tylerbutler) [inline build-tools/packages/changelog-generator/src/test/tsconfig.json:8]:
```suggestion
		"outDir": "../../dist/test",
```

**Review by RishhiB — APPROVED** [review]
Looks good, just gotta get CI to pass

**Pull request merged**